File: src/test/java/com/baulsupp/oksocial/i9n/TestCompletionVariableCache.java
package com.baulsupp.oksocial.i9n;

import com.baulsupp.oksocial.completion.CompletionVariableCache;
import com.google.common.collect.Maps;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class TestCompletionVariableCache implements CompletionVariableCache {
  private Map<String, List<String>> cache = Maps.newConcurrentMap();

  @Override public Optional<List<String>> get(String service, String key) {
    return Optional.ofNullable(cache.get(service + "-" + key));
  }

  @Override public void store(String service, String key, List<String> values) {
    cache.put(service + "-" + key, values);
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/i9n/TestCompletionVariableCache.kt
package com.baulsupp.oksocial.i9n

import com.baulsupp.oksocial.completion.CompletionVariableCache
import java.util.concurrent.ConcurrentHashMap

class TestCompletionVariableCache : CompletionVariableCache {
  private val cache = ConcurrentHashMap<String, List<String>>()

  override fun get(service: String, key: String): List<String>? {
    return cache["$service-$key"]
  }

  override fun set(service: String, key: String, values: List<String>) {
    cache["$service-$key"] = values
  }
}
